NEW: Add prepare() method to DoliDB class to support prepared SQL statements

R0ck3n opened the pull request:
# NEW: Add prepare() method to DoliDB class

This PR adds support for prepared SQL statements in Dolibarr by implementing a `prepare()` method in the `DoliDB` abstract class.

## Purpose
Prepared statements help mitigate SQL injection risks and can improve performance when executing similar queries multiple times.

## Implementation
- Added a `prepare($sql)` method in `DoliDB`, which delegates to the native `prepare()` method of the underlying DB handler (e.g., MySQLi).
- Handles error assignment (`$this->lasterror`) if the statement fails.

## Usage example
```php
global $db;

$sql = "SELECT rowid, login, lastname, firstname FROM " . MAIN_DB_PREFIX . "user WHERE rowid = ?";

// Préparation
$stmt = $db->prepare($sql);

if ($stmt === false) {
    echo "Erreur de préparation : " . $db->lasterror;
    exit;
}

// Lier le paramètre (ici 'i' pour int, et on passe 1 pour rowid)
$rowid = 1;
$stmt->bind_param("i", $rowid);

// Exécution
if (!$stmt->execute()) {
    echo "Erreur à l'exécution : " . $stmt->error;
    exit;
}

// Récupération du résultat
$result = $stmt->get_result();

if ($result && $result->num_rows > 0) {
    $user = $result->fetch_assoc(); // ou fetch_object() si tu préfères un objet
    var_dump($user);
} else {
    echo "Aucun utilisateur trouvé avec rowid = 1.";
}

// Nettoyage
$stmt->close();
```

R0ck3n committed:
feat:add prepared function to DoliDB class

R0ck3n committed:
docs: add comment on sql param in prepare method

marc-dll committed:
FIX: accountancy general ledger: bad handling of hook return (#34029)

* FIX: accountancy general ledger: bad handling of hook return

* FIX: accountancy balance: bad handling of hook return

eldy committed:
Update changelog

eldy committed:
Merge branch '21.0' of git@github.com:Dolibarr/dolibarr.git into 21.0

eldy committed:
Increase period for alert to 6 monthes.

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge branch '21.0' of git@github.com:Dolibarr/dolibarr.git into 22.0

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix css

eldy committed:
Better management of how we store localtax vat rate for customer price.

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

frederic34 committed:
fix CI (Travis) (#35099)

* fix test

* fix test

* fix test

* fix test

* fix test

* fix

* fix

* fix

eldy committed:
FIX #35098

lvessiller-opendsi committed:
FIX set global mysoc and load langs in API access (#35041)

* FIX set global mysoc and load langs in API access

* FIX reset mysoc and langs only if entity of API has changed

eldy committed:
CSS

SylvainLegrand committed:
Fix supplier invoice API (#35100)

* Fix supplier invoice API

We can't modify mode_reglement_id from API on supplier invoice !

* Update fournisseur.facture.class.php

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

eldy committed:
FIX #35105

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
NEW More index for memberships table

aspangaro committed:
FIX Treasury accountancy - Accounting errors for members (#35097)

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

github-actions[bot] committed:
PHPStan > Update baseline (#35102)

Co-authored-by: Dolibot <dolibarr-bot@users.noreply.github.com>
Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

frederic34 committed:
avoid untranslated dol_print_error (#35104)

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

defrance committed:
fix translation of $fields on interventional (#35106)

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

frederic34 committed:
Update box_last_modified_ticket.php (#35111)

frederic34 committed:
translations fields (#35112)

eldy committed:
FIX #34981

eldy committed:
Merge branch '21.0' of git@github.com:Dolibarr/dolibarr.git into 21.0

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into 22.0

eldy committed:
FIX #35071

eldy committed:
Fix reposition

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

frederic34 committed:
Translation fields (#35114)

* translations fields

* translations fields

eldy committed:
FIX #35115

Daoud-mohamed committed:
Prepare SEPA payments for supplier invoices via Stripe (#35078)

* preparation for supplier invoice payment via SEPA Stripe

* use stripe cynematic

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

eldy committed:
Prepare 21.0.3

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into 22.0

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

W1W1-M committed:
Fixed intervention line duration add (#35117)

frederic34 committed:
fix box_last_modified_ticket.php (#35121)

eldy committed:
NEW add a limit to avoid too many answer in agenda view. Add warning if
limit has been reached.

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

frederic34 committed:
fetch greatest tms (#35122)

* fetch greatest tms

* fetch greatest tms

* fetch greatest tms

* fetch greatest tms

* fetch greatest tms

* fetch greatest tms

* fetch greatest tms

* fetch greatest tms

* fetch greatest tms

* fetch greatest tms

eldy committed:
Add icon for apple-touch-icon.png

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix message

eldy committed:
Fix message

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into 22.0

frederic34 committed:
fetch greatest tms (#35123)

* fetch greatest tms

* fetch greatest tms

* fetch greatest tms

* fetch greatest tms

lemmings67 committed:
fix #35118 (#35119)

OpenID doesn't work when $dolibarr_main_authentication contains "ldap"

Co-authored-by: Pascal EISELE <peisele@appligos.fr>

eldy committed:
Fix bad index

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into develop

github-actions[bot] committed:
PHPStan > Update baseline (#35134)

Co-authored-by: Dolibot <dolibarr-bot@users.noreply.github.com>

frederic34 committed:
wip workstation tms (#35126)

* wip workstation tms

* wip workstation tms

* setup

* setup

* setup

* setup

* setup

eldy committed:
FIX error when using a code too large in dictionary

frederic34 committed:
fetch greatest tms for members (#35133)

* fetch greatest tms for members

* fetch greatest tms for members

thersane-john committed:
fix experimental intuitive select (#35131)

FHenry committed:
new: add fitid columnin llx_bank_import (#35128)

ro78 committed:
FIX: #33911 active services are reported as expired on the homepage (#35125)

eldy committed:
Trans

eldy committed:
Clean setup of antivirus

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

github-actions[bot] committed:
PHPStan > Update baseline (#35146)

Co-authored-by: Dolibot <dolibarr-bot@users.noreply.github.com>

Thatoo committed:
auto_subscription_member_form_types_nature (#35033)

hregis committed:
FIX wrong country code (#35152)

eldy committed:
FIX setup of default doc template for supplier proposal when none.

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into 22.0

eldy committed:
Fix default order in list of supplier proposals

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
FIX changin status on proposal need a refresh of page to be visible

eldy committed:
FIX default doc template on supplier proposal

eldy committed:
FIX Security when using feature Advanced Target of emailing (hidden in
v21, default in v22+). Possible SQL injection by users with permissions
to make mass emailing. Reported by Simon Perdrisat.

eldy committed:
Merge branch '21.0' of git@github.com:Dolibarr/dolibarr.git into 22.0

aspangaro committed:
FIX Missing sentence part (#35144)

frederic34 committed:
fetch greatest tms for user and fix user creation and modification (#35145)

* fetch greatest tms for user and fix user creation and modification

* fetch greatest tms for user and fix user creation and modification

* fetch greatest tms for user and fix user creation and modification

* fetch greatest tms for user and fix user creation and modification

* fetch greatest tms for user and fix user creation and modification

* fetch greatest tms for user and fix user creation and modification

frederic34 committed:
clean code (#35148)

* clean code

* fix

* fix

* fix

eldy committed:
Fix CI

sonikf committed:
fix #34872 (#35150)

* Remove fk_product check brakes addition of free product lines

* Handle potential null for detail_children array

frederic34 committed:
clean code (#35151)

* clean code

* Update mod_barcode_thirdparty_standard.php

* Update mod_barcode_product_standard.php

sonikf committed:
fix grammar (#35158)

* Update company birth date label to establishment date

* Simplify default bank account description

* Fix wording in discount split labels

* fix grammar

* Fix typo

frederic34 committed:
fix Variable $statut might not be defined (#35161)

comaiteseb committed:
FIX #35154 - Massaction createbills in reception list don't mind about order and/or thirdparty currency (#35156)

* FIX#35154

Massaction createbills in reception list don't mind about order and/or thirdparty currency

* FIX#35154

* FIX#35154

* Delete white space line 406

* Correcting typing error

lvessiller-opendsi committed:
FIX missing quick edit for extrafields (baclport commit 4fc66c6) (#35160)

eldy committed:
Fix warning

eldy committed:
Fix warning

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

bchanudet committed:
FIX #35147 SQL Error on Beluga Export when ExpenseReport is enabled (#35149)

* Fix SQL error when ExpenseReport module is enabled (#35147)

* fix date filtering that prevents data to be displayed

* add date and amount for expense reports

* Update pdf_beluga.modules.php

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

eldy committed:
Fix HTML escaping

eldy committed:
Comment

frederic34 committed:
add doc (#35166)

* add doc

* add doc

* add doc

frederic34 committed:
clean code (#35167)

* clean code

* clean code

frederic34 committed:
add doc (#35168)

* add doc

* add doc

* add doc

* add doc

* add doc

* add doc

* add doc

altairis-noe committed:
NEW: generate renewal proposal for contracts (#35120)

* NEW: generate renewal proposal for contracts

* Update card.php

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

eldy committed:
NEW Button to create a proposal and sale order from a contract always on

eldy committed:
FIX we saw links to object on private not just after a save.

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into 22.0

eldy committed:
Prepare 21.0.4

eldy committed:
Merge branch '21.0' of git@github.com:Dolibarr/dolibarr.git into 21.0

eldy committed:
Doc

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into 22.0

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix position in changelog

eldy committed:
Trans

eldy committed:
FIX fallback of "from" email when sending email in member module

eldy committed:
Clean code

eldy committed:
Better virtual host file

eldy committed:
Sec: Update doc to trigger alert of vulnerability fix in commit
bb0974add9cb746c2f1723a239bbaf50561251c6

eldy committed:
Merge branch '21.0' of git@github.com:Dolibarr/dolibarr.git into 22.0

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into develop

frederic34 committed:
clean code (#35183)

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

eldy committed:
Fix CI

eldy committed:
Merge branch '21.0' of git@github.com:Dolibarr/dolibarr.git into 22.0

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

frederic34 committed:
add doc (#35182)

mdeweerd committed:
Fix: Fix typo in user_modifcation_id (#35179)

# Fix: Fix typo in user_modifcation_id

Thes fixes the modification id initialisation.

github-actions[bot] committed:
PHPStan > Update baseline (#35176)

Co-authored-by: Dolibot <dolibarr-bot@users.noreply.github.com>

frederic34 committed:
clean code (#35175)

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

frederic34 committed:
fix CI v22 (#35173)

frederic34 committed:
clean code (#35170)

* clean code

* clean code

frederic34 committed:
add doc (#35185)

* add doc

* add doc

* add doc

* add doc

* add doc

* add doc

atm-jonathan committed:
Fix load extrafields list task (#35135)

* load extrafieds task

* load extrafieds task

* restore commentaire

---------

Co-authored-by: Jonathan <x@x.x>

eldy committed:
FIX #35142 FIX #35143

eldy committed:
Merge branch '21.0' of git@github.com:Dolibarr/dolibarr.git into 21.0

eldy committed:
Merge branch '21.0' of git@github.com:Dolibarr/dolibarr.git into 22.0

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into 22.0

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
FIX color of late warning for nb of bank account reconciliation lines

eldy committed:
FIX color of late warning for nb of bank account reconciliation lines

eldy committed:
Fix CI warning

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into develop

mdeweerd committed:
Qual: Update phan baseline (#35174)

# Qual: Update phan baseline

Remove exceptions that were fixed

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

atm-lucasmantegari committed:
FIX - Return False with return code 200 on API (#35172)

* Return false in API

* Update api_contracts.class.php

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

zahwyrix committed:
FIX Quick search sellist extrafield (#35070)

* FIX Quick search sellist extrafield

* Update functions.lib.php

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

atm-jpb committed:
fix bug div added without value (#35178)

Co-authored-by: jpb <jean-pascal.boudet@atm-consulting>
Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

jyhere committed:
NEW: add combining characters (accents, cedilla...) codes in dol_string_unaccent() (#35130)

* NEW: add combining characters (accents, cedilla...) codes in dol_string_unaccent()

* NEW: add combining characters (accents, cedilla...) codes in dol_string_unaccent()

hermans committed:
Improve setOrigin on Receipt for line_id_object_src & line_id_object_origin (#35091)

In Receipt validate, close... we have setOrigin that not set data line_id_object_src & line_id_object_origin, that variable is very important when we working with custom hook. I checked also on shipment it already implemented.

frederic34 committed:
clean code (#35189)

* clean code

* clean code

aspangaro committed:
NEW Accountancy - Add hook on export filename (#35188)

* NEW Accountancy - Add hook on export filename

* FIX CI

eldy committed:
Clean code

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

frederic34 committed:
add a combining character in dol_string_unaccent test (#35153)

* add a combining character in dol_string_unaccent test

* fix

pratushraj committed:
#35192 (#35193)

mapiolca committed:
FIX MULTICOMPANY Error (#35195)

Without it, We have a "error 500" on this tab when Multicomany is active.

eldy committed:
Clean code

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Clean code

frederic34 committed:
fix doc (#35196)

eldy committed:
Fix warning

github-actions[bot] committed:
PHPStan > Update baseline (#35200)

Co-authored-by: Dolibot <dolibarr-bot@users.noreply.github.com>

frederic34 committed:
clean code (#35180)

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* fix display

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

frederic34 committed:
clean code (#35191)

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

* clean code

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

bchanudet committed:
FIX #35181 Cancelled purchase orders now shown as such in Project Overview (#35186)

* exclude "cancelled" purchase orders from total

* show cancelled purchase order as cancelled

* show cancelled order by default, as the visibility button is "cancelled shown" by default

eldy commented on htdocs/core/db/DoliDB.class.php:
```diff
@@ -459,4 +459,25 @@ public function getRows($sql)
+			return false;
+		}
+
+		$stmt = $this->db->prepare($sql);
```
I think the implementation with
$stmt = $this->db->prepare($sql);

should not be into the DoliDB.class.php but into the DoliDBMysqli class. Because method ->prepare does not exists in postgresql.
So we can add the 
public function prepare($sql)
with empty content into DoliDB.class.php
and the complete implementation into class  DoliDBMysqli  and DoliDBPgsql

mdeweerd committed:
Qual: remove fixed phan notice from exceptions (#35201)

# Qual: remove fixed phan notice from exceptions

The pdf.lib.php has one type of exception less.

mpellegrin committed:
FIX #35007 (#35010)

* Fix #35007

* Update pay.php

* Update pay.php

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

eldy committed:
The signature of the Sale order ->cancel() method and shipment
->cancel() has been modified to introduce the $user param like for other
methods that modify a status.

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Try CI fix

Hystepik committed:
Fix impossibility to validate objects (#35202)

* Fix impossibility to validate objects

* fix pre commit

---------

Co-authored-by: Lucas Marcouiller <lmarcouiller@dolicloud.com>

eldy committed:
Doc

eldy committed:
Doc

kkhelifa-opendsi committed:
FIX: Add security test for show terminal selection if no terminal selected when invoice.php is call (#34717)

* FIX: Add security test for show terminal selection if no terminal selected when invoice.php is call

* Modification to take into account the value saved in the cookie

* Update invoice.php

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

eldy committed:
NEW Can view/list/edit the dispute status of an invoice

eldy committed:
Fix phpunit

frederic34 committed:
fix CI (#35204)

eldy committed:
Fix trans

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
FIX Restore decimal quantity for stock correction/transfer

eldy committed:
Merge branch '20.0' of git@github.com:Dolibarr/dolibarr.git into 20.0

eldy committed:
Merge branch '20.0' of git@github.com:Dolibarr/dolibarr.git into 21.0

eldy committed:
Merge branch '21.0' of git@github.com:Dolibarr/dolibarr.git into 22.0

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into 22.0

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into
develop

eldy committed:
Fix CI

eldy committed:
Fix CI

eldy committed:
Enhance template to allow hook printFieldListWhere

frederic34 committed:
clean test code (#35205)

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

eldy committed:
QUAL: Removed no more used deprecated code

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Clean code between subprice/pu_ht and subprice_ttc/pu_ttc

frederic34 committed:
fix parameters (#35209)

frederic34 committed:
clean code (#35210)

eldy committed:
Clean code. Show how to use TRIGGER_PREFIX in modulebuilder template.

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix CI warnings

eldy committed:
Try to fix ci

eldy committed:
Fix CI

SylvainLegrand committed:
Fix discount applied 2 times on contract (#35219)

on update line the discount was apply 2 times !
no problem on insert line.

Already solved on v22

eldy committed:
Fix CI warnings

eldy committed:
Fix CI

frederic34 committed:
fix CI (#35215)

* fix CI

* Update fournisseur.commande.class.php

bohdanpotuzhnyi committed:
support RFC5789 (patch) (#35213)

kkhelifa-opendsi committed:
FIX: Correct the calculation of the amount of the current period between the period provided (#35083)

eldy committed:
Merge branch '18.0' of git@github.com:Dolibarr/dolibarr.git into 19.0

eldy committed:
Merge branch '20.0' of git@github.com:Dolibarr/dolibarr.git into 20.0

eldy committed:
Merge branch '20.0' of git@github.com:Dolibarr/dolibarr.git into 21.0

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into 22.0

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Avoid phpstan error

eldy committed:
Avoid phpstan warning

eldy committed:
Fix CI

mdeweerd committed:
Qual: Update phan baseline (#35214)

# Qual: Update phan baseline

Update phan baseline by removing fixed exceptions

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

frederic34 committed:
fix modulebuilder editorconfig (#35221)

to be the same than in dolibarr root

frederic34 committed:
add value to make update of object easier (#35222)

eldy committed:
Fix warning

eldy committed:
Fix warning

eldy committed:
Merge branch '20.0' of git@github.com:Dolibarr/dolibarr.git into 21.0

eldy committed:
Merge branch '21.0' of git@github.com:Dolibarr/dolibarr.git into 22.0

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Remove warnings

eldy committed:
Fix warning

eldy committed:
Fix CI

eldy committed:
Rename github workflow files

eldy committed:
Fix CI

pratushraj committed:
#35216 (#35217)

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

eldy committed:
Merge branch '21.0' of git@github.com:Dolibarr/dolibarr.git into 22.0

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix label of tooltip not complete

eldy committed:
Fix label of tooltip not complete

vmaury committed:
Fix 24338 In expense report list, all expense reports are a warning icon (#35225)

Co-authored-by: vmaury <vmaury@vmaury-Lafite-Pro-16-AMD>

eldy committed:
Merge branch '21.0' of git@github.com:Dolibarr/dolibarr.git into 22.0

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix warning

eldy committed:
Fix warning

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

frederic34 committed:
fix CI v22 (#35227)

* fix CI v22

* Update time.php

* Update card.php

* Update card.php

* Update card.php

* Update card.php

* Update card.php

* Update card.php

* Update card.php

* Update card.php

eldy committed:
Merge branch '22.0' of git@github.com:Dolibarr/dolibarr.git into develop

frederic34 committed:
fix broken workflows (#35226)

* fix broken workflows

* Update ci-on-push.yml

* Update fournisseur.commande.class.php

* clean code

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

frederic34 committed:
add reminder by sms (#35236)

* add reminder by sms

* fix url dolistore

mdeweerd committed:
Qual: Update phan baseline (#35237)

# Qual: Update phan baseline

eldy committed:
Fix warning

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix CI

eldy committed:
Better name for var

pratushraj committed:
#35230 (#35231)

* #35230

* Update massstockmove.php

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

eldy committed:
Clean code

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Add link to category edit page from picto

eldy committed:
CSS

frederic34 committed:
NEW Add sms reminder in reminder of agenda events (#35239)

* add reminder by sms

* fix url dolistore

* add cron task to send sms reminder

* add cron task to send sms reminder

* add cron task to send sms reminder

* add cron task to send sms reminder

* Update sms.php

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

github-actions[bot] committed:
PHPStan > Update baseline (#35240)

Co-authored-by: Dolibot <dolibarr-bot@users.noreply.github.com>

eldy committed:
NEW Can sort on employee in holiday balance. Add link to go to history.

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

R0ck3n committed:
feat(db): add support for prepared statements in PostgreSQL driver

R0ck3n committed:
Merge branch 'feat/prepared_sql' of https://github.com/R0ck3n/dolibarr into feat/prepared_sql
